util/settings/settings_test.go

```diff
@@ -420,3 +420,52 @@ func Test_validateExternalURL(t *testing.T) {
 		})
 	}
 }
+
+func TestSecretKeyRef(t *testing.T) {
+	data := map[string]string{
+		"dex.config": `
+      connectors:
+        # GitHub example
+        - type: github
+          id: github
+          name: GitHub
+          config:
+            clientID: aabbccddeeff00112233
+            clientSecretSecretRef:
```

Comment:
Argo CD already supports referencing values from `argocd-secret` Secret. E.g. `$dex.github.clientSecret`. Can we keep using similar syntax for consistency. For example: `$<secretName>:<key>`. where `<secretName>` is optional secret name. 

In this case secret sample config would looks like following:

```
        config:
            clientID: aabbccddeeff00112233
            clientSecret: $google:clientSecret
```


It is safe to use `:` as a delimiter since it cannot be used in secret name/secret data key.

Reply:
I can understand the need for consistency. Here, I just wanted to be closer to `SecretKeySelector` type that exists in Kubernetes (cf. https://github.com/kubernetes/api/blob/master/core/v1/types.go#L1984) or in FluxCD's helm-operator (cf. https://github.com/fluxcd/helm-operator/blob/68de5288b733c42a316e03f9c01e7505cd6985b3/pkg/apis/helm.fluxcd.io/v1/types.go#L48).

I'm relatively new in this world, so I do not really have arguments against your proposal.

(And we do `ArcgoCD` here, not `FluxCD` :smile: )